test/replay/btcstaking_test.go

```diff
@@ -351,3 +351,173 @@ func TestStakingAndFinalizingMultipleBlocksAtOnce(t *testing.T) {
 		require.Equal(t, bl.Finalized, true)
 	}
 }
+
+func TestSlashingandFinalizingBlocks(t *testing.T) {
+	t.Parallel()
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	driver := NewBabylonAppDriverTmpDir(r, t)
+	driver.GenerateNewBlock()
+
+	scenario := NewStandardScenario(driver)
+	scenario.InitScenario(6, 1) // 6 finality providers
+
+	numBlocksInTest := uint64(10)
+	lastVotedBlkHeight := scenario.FinalityFinalizeBlocks(scenario.activationHeight, numBlocksInTest)
+
+	indexSlashFp1 := 1
+	indexSlashFp2 := 2
+	jailedFp1 := scenario.finalityProviders[indexSlashFp1]
+	jailedFp2 := scenario.finalityProviders[indexSlashFp2]
+
+	bl := driver.GetIndexedBlock(lastVotedBlkHeight)
+	require.Equal(t, bl.Finalized, true)
+
+	// 2 fps not voting
+	for i := uint64(0); i < numBlocksInTest; i++ {
+		lastVotedBlkHeight++
+		for i, fp := range scenario.finalityProviders {
+			if i != indexSlashFp1 {
+				fp.CastVote(lastVotedBlkHeight)
+			}
+		}
+
+		driver.GenerateNewBlock()
+
+		bl := driver.GetIndexedBlock(lastVotedBlkHeight)
+		require.Equal(t, bl.Finalized, true)
+	}
+
+	scenario.finalityProviders[indexSlashFp1].SendSelectiveSlashingEvidence()
+	scenario.finalityProviders[indexSlashFp2].SendSelectiveSlashingEvidence()
+
+	driver.GenerateNewBlock()
+
+	slashedFp1 := driver.GetFp(*jailedFp1.BTCPublicKey())
+	require.True(t, slashedFp1.IsSlashed())
+	slashedFp2 := driver.GetFp(*jailedFp2.BTCPublicKey())
+	require.True(t, slashedFp2.IsSlashed())
+
+	driver.GenerateNewBlock()
+
+	bl = driver.GetIndexedBlock(lastVotedBlkHeight)
+	require.Equal(t, bl.Finalized, true)
+
+	require.Len(t, driver.GetActiveFpsAtCurrentHeight(t), 4)
+}
+
+func TestActivatingDelegationOnSlashedFp(t *testing.T) {
```

Comment:
I think this scenario the ordering should be (after initializing standard scenario):
1. Create pre-approval deleagtion to one fp
2. Send cov signatures for this delegaiton
3. Slash the fp
4. Now try to activate delegation. This should succeed, but FP should remaing to be slashed and not able to vote